Example/UserSettingsTableViewController.swift

```diff
@@ -0,0 +1,108 @@
+        tableView.deselectRow(at: indexPath, animated: false)
+    }
+
+    func setProfileImage(image: UIImage?) {
+        guard let profileImage = image else {
+            editPhotoButton.setTitle("Add Photo", for: .normal)
+            return
+        }
+        editPhotoButton.setTitle("Edit", for: .normal)
+        self.imageView.image = profileImage
+        self.userProfileImage = profileImage
+    }
+
+    @IBAction func showPhotoLibrary() {
+        let imagePickerController = UIImagePickerController()
+        imagePickerController.sourceType = .savedPhotosAlbum
+        imagePickerController.allowsEditing = true
+        imagePickerController.delegate = self
+        present(imagePickerController, animated: true)
+    }
+
+    @IBAction func modifyProfileSettings() {
+        if modifyProfileSettingsButton.title == "Edit" {
+            modifyProfileSettingsButton.title = "Save"
+        } else {
+            if !saveProfileSettings() {
+                return
+            }
+            modifyProfileSettingsButton.title = "Edit"
+        }
+        displayNameTextField.isEnabled = !displayNameTextField.isEnabled
+        editPhotoButton.isEnabled = !editPhotoButton.isEnabled
+        editPhotoButton.isHidden  = !editPhotoButton.isHidden
+        self.displayNameTextField.becomeFirstResponder()
+    }
+
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey: Any]) {
+        picker.dismiss(animated: true)
+
+        guard let image = info[.editedImage] as? UIImage else {
+            print("User cancelled the operation")
+            return
+        }
+        setProfileImage(image: image)
+    }
+
+    func saveProfileSettings(forKey key: String = "ProfileImage") -> Bool {
+        let name = displayNameTextField.text?.trimTrailingWhitespaces()
+        guard let userDisplayName = name, !userDisplayName.isEmpty else {
+            self.displayAlert(title: NSLocalizedString("error_title", comment: ""), message: NSLocalizedString("error_user_profile_name_not_found", comment: ""))
+            return false
+        }
+
+        if let data = try? PropertyListEncoder().encode(UserProfileModel(displayName: userDisplayName, profileImage: self.userProfileImage)) {
+            UserDefaults.standard.set(data, forKey: key)
+            return true
+        }
+        return false
+    }
+
+    func retrieveProfileSettings(key: String = "ProfileImage") -> UIImage? {
+        if let userProfile = UserDefaults.standard.data(forKey: key) {
+            let userProfileData = try? PropertyListDecoder().decode(UserProfileModel.self, from: userProfile)
+            self.displayNameTextField.text = userProfileData?.displayName
+            guard let imageData = userProfileData?.profileImageData else {
+                return nil
+            }
+            return UIImage(data: imageData)
```

Comment:
I just got an update and it's looking like the image will be provided as a URL. Is it possible to provide a URI to the JavaScript side for the image that was chosen by the picker? I think we specifically designed it so the JS side can't access anything else on the file system, right?

So maybe this will need to be changed to accept an image URL instead...

Reply:
We can change it to URI, that URI will be coming from `tmp` path. If that is ok ??

Reply:
@ts-leojoseph-r But will the Mini App be able to access a URI from `tmp` path? I thought that a Mini App can't access a file outside of its own files.

I just thought of another idea: we could convert the image to base64 and pass it as a data URI. See here: https://stackoverflow.com/a/25516291 . What do you think?

Reply:
Ah, I think this would work for sure. I will update the code

Reply:
Done